### PR TITLE
Merging velero features to application-mobility branch

### PR DESCRIPTION
**Pull request by kumarkgosa**
Adding velero features
1. Use volume snapshots
2. Cleaup crds
and make unit-tests work to merge in our unified branch for application-mobility

### CODE REVIEW

File: main.go
Comment (jooseppi-luna) [inline main.go:1]: Do we need these changes?

File: pkg/modules/application_mobility.go [hunk 1]
```diff
@@ -407,51 +458,79 @@ func getVelero(op utils.OperatorConfig, cr csmv1.ContainerStorageModule) (string
...
-		}
+	appMob, err := getAppMobilityModule(cr)
+	if err != nil {
+		return YamlString, err
 	}
 
-	return false
-}
-
-// IsAppMobilityComponentEnabled - check if Application Mobility componenets are enabled
```
Comment (jooseppi-luna) [inline pkg/modules/application_mobility.go:426]: comment needs to match function

File: pkg/modules/application_mobility.go [hunk 2]
```diff
@@ -407,51 +458,79 @@ func getVelero(op utils.OperatorConfig, cr csmv1.ContainerStorageModule) (string
...
+	return YamlString, nil
 }
 
-// IsAppMobilityComponentEnabled - check if Application Mobility componenets are enabled
-func IsdeployResticEnabled(ctx context.Context, instance csmv1.ContainerStorageModule, r ReconcileCSM, mod csmv1.ModuleType, componentType string) bool {
-	appMobilityEnabled, appmobility := IsModuleEnabled(ctx, instance, mod)
-	if !appMobilityEnabled {
-		return false
+func getCleanupcrds(op utils.OperatorConfig, cr csmv1.ContainerStorageModule) (string, error) {
+	YamlString := ""
```
Comment (jooseppi-luna) [inline pkg/modules/application_mobility.go:504]: local variables (not being exported outside of the function) need to be camelcase with lowercase first letter

File: pkg/modules/application_mobility.go [hunk 3]
```diff
@@ -407,51 +458,79 @@ func getVelero(op utils.OperatorConfig, cr csmv1.ContainerStorageModule) (string
...
+	buf, err := os.ReadFile(filepath.Clean(CleanupCrdsPath))
+	if err != nil {
+		return YamlString, err
 	}
 
-	for _, c := range appmobility.Components {
-		if c.Name == componentType && *c.Enabled {
-			return true
+	YamlString = string(buf)
+	Velero_NS := ""
```
Comment (jooseppi-luna) [inline pkg/modules/application_mobility.go:518]: same comment here

File: samples/application-mobility/csm_application_mobility_v030.yaml [hunk 4]
```diff
@@ -51,8 +50,9 @@ spec:
         image: velero/velero:v1.8.1
         imagePullPolicy: IfNotPresent
         enabled: true
-        #use-volume-snapshots: false
-        #cleanUpCRDs: false
+        use-volume-snapshots: true
```
Comment (jooseppi-luna) [inline samples/application-mobility/csm_application_mobility_v030.yaml:53]: this should probably be "useVolumeSnapshots" just to be consistent with everything else

**Review by jooseppi-luna — CHANGES_REQUESTED** [review]
looks good overall just some minor comments -- nice work!

File: main.go
Comment (kumarkgosa) [reply main.go:1]: Yup, It's needed. Velero installation fails without main.go changes. I have tested

File: pkg/modules/application_mobility.go (hunk 1 above)
Comment (kumarkgosa) [reply pkg/modules/application_mobility.go:426]: done

File: samples/application-mobility/csm_application_mobility_v030.yaml (hunk 4 above)
Comment (kumarkgosa) [reply samples/application-mobility/csm_application_mobility_v030.yaml:53]: done!

File: pkg/modules/application_mobility.go (hunk 2 above)
Comment (kumarkgosa) [reply pkg/modules/application_mobility.go]: done

**Review by mbasha-dell — APPROVED** [review]

**Pull request merged**